wgpu-native/src/device.rs

```diff
@@ -572,14 +600,25 @@ pub extern "C" fn wgpu_queue_submit(
     // now prepare the GPU submission
     let fence = device.raw.create_fence(false).unwrap();
     {
+        let swap_chain_guard = HUB.swap_chains.read();
+        let wait_semaphores = swap_chain_links
+            .into_iter()
+            .map(|link| {
+                //TODO: check the epoch
+                let sem = &swap_chain_guard
+                    .get(link.swap_chain_id.0)
+                    .frames[link.image_index as usize]
+                    .sem_available;
+                (sem, hal::pso::PipelineStage::COLOR_ATTACHMENT_OUTPUT)
+            });
         let submission =
             hal::queue::Submission::<_, _, &[<back::Backend as hal::Backend>::Semaphore]> {
                 //TODO: may `OneShot` be enough?
                 command_buffers: command_buffer_ids
                     .iter()
                     .flat_map(|&cmb_id| &command_buffer_guard.get(cmb_id).raw),
-                wait_semaphores: Vec::new(),
-                signal_semaphores: &[],
+                wait_semaphores,
+                signal_semaphores: &[], //TODO: signal `sem_present`?
```

Comment:
Could you explain this comment a bit more to clarify the synchronization here?

Reply:
It's not used because we are waiting for the frame fence in `get_next_texture`. So technically `sem_present` could be removed

Reply:
Alright, is there a reason to still keep it then? I guess we might need it later?